subsys/usb/usb_device.c

```diff
@@ -937,10 +943,51 @@ static void usb_register_status_callback(usb_dc_status_callback cb)
 	usb_dev.status_callback = cb;
 }
 
+static int foreach_ep(int (* endpoint_callback)(const struct usb_ep_cfg_data *))
+{
+	size_t size = (__usb_data_end - __usb_data_start);
+
+	for (size_t i = 0; i < size; i++) {
+		struct usb_cfg_data *cfg = &__usb_data_start[i];
+		struct usb_ep_cfg_data *ep_data = cfg->endpoint;
+
+		for (u8_t n = 0; n < cfg->num_endpoints; n++) {
+			int ret;
+
+			ret = endpoint_callback(&ep_data[n]);
+			if (ret < 0) {
+				return ret;
+			}
+		}
+	}
+
+	return 0;
+}
+
+static int disable_interface_ep(const struct usb_ep_cfg_data *ep_data)
+{
+	u8_t ep = ep_data->ep_addr;
+
+	if (ep | !0x80) {
```

Comment:
What it is supposed to do?

Reply:
Sorry, my bad, removed that

Reply:
In general, it is good not to allow to disable EP0 and I suppose it was relating to that issue. I am not sure if the underlying library does take care of this?

Reply:
That endpoint table contains only non control endpoints, so we do not need this (incorrect) check.